### PR TITLE
add delete trip functionality

### LINKED ISSUE #190: Add delete trip functionality
User should be able to click a delete button in the trip view screen and have a pop up dialog to confirm the deletion.

### PR DESCRIPTION
**Pull request by mirceagavrizi**
## Summary

This PR introduces the possibility to delete a trip. I will attach a screen recording to present the process which is built in a user friendly mode.

## Changes

- **Screens/UI:** The user can now delete a trip
- **Bug Fixes/Improvements:** Addresses #190 

## Checklist

- [x] This PR resolves #190 
- [x] Tests have been added for new functionality.
- [x] Screenshots or UI changes have been attached (if applicable).
- [x] Documentation has been updated (if applicable).

##  Testing

- **Manual Testing:**
    - [x] Tested on emulator / physical device.
- **Unit Tests**
    - [x] Added units tests for this

##  Related Issues/Tickets

Closes #190 

## Screenshots (if applicable)

https://github.com/user-attachments/assets/61403abf-a1c7-4fe4-9dcc-aaa9eb8fc340

### CODE REVIEW

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt [hunk 1]
```diff
@@ -209,8 +220,46 @@ fun TripItem(
...
+  // Confirmation Dialog
+  if (showDialog) {
+    AlertDialog(
+        onDismissRequest = { showDialog = false },
+        title = { Text(text = "Remove Trip") },
+        text = { Text("Are you sure you want to remove \"${trip.name}\" from your trips?") },
+        confirmButton = {
+          TextButton(
+              onClick = {
+                tripsViewModel.deleteTripById(trip.id)
```
Comment (apana03) [inline app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:255]: Users do not receive any visual feedback upon successfully deleting a trip, we should use the OnSuccess parameter of the deleteTripById to display a message saying that we indeed deleted that trip.

Comment (apana03) [reply app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:255]: Also, what happens if the deleteTripbyId fails ? Here the deletion action assumes success without verifying the result from deleteTripById. We should look into error handling.

**Review by apana03 — CHANGES_REQUESTED** [review]
@mirceagavrizi Nice Work! Thank you for implementing this feature. Had only 2 small comments regarding the usage of the viewModel.

File: app/src/main/java/com/android/voyageur/ui/overview/Overview.kt (hunk 1 above)
Comment (mirceagavrizi) [reply app/src/main/java/com/android/voyageur/ui/overview/Overview.kt:258]: You're right, it's better to take into account all cases. Also, for better UX, I added a toast to be displayed when the trip is successfully deleted.

**Review by apana03 — APPROVED** [review]
LGTM! Thank you for taking into account the previous comments.

**Pull request merged**